File: Python/Problems/Problems/problem1910.py
from typing import List

from .problem import Problem


class Problem1910(Problem):
    def test(self) -> bool:
        s = "daabcbaabcbc"
        part = "abc"

        expected = "dab"

        result = self.removeOccurrences(s, part)

        print(result)

        return result == expected

    def removeOccurrences(self, s: str, part: str) -> str:
        index = s.find(part)
        while index != -1:
            s = s[0:index] + s[index + len(part):]
            index = s.find(part)

        return s

File: C++/Problems/Problems/problem_1910.cpp
#include "problem_1910.h"

bool problem_1910::test() {
    std::string s = "daabcbaabcbc";
    std::string part = "abc";

    std::string expected = "dab";

    auto result = removeOccurrences(s, part);

    std::cout << result << std::endl;

    return result == expected;
}

std::string problem_1910::removeOccurrences(std::string s, std::string part) {
    auto index = s.find(part);
    while (index != -1)
    {
        s = s.substr(0, index) + s.substr(index + part.size());
        index = s.find(part);
    }

    return s;
}
